src/QuantumComputation.cpp

```diff
@@ -830,12 +828,12 @@ namespace qc {
 
                 // base case -> determine the basis state from the measurement and safe the probability
                 if (measurements.size() == nclassics - 1) {
-                    std::size_t idx0 = 0;
-                    std::size_t idx1 = 0;
+                    std::uint_least64_t idx0 = 0;
+                    std::uint_least64_t idx1 = 0;
```

Comment:
Keep these as `std::size_t`? The probability vector takes a `std::size_t` as key.